### PR TITLE
Basic SMT tests.

### PR DESCRIPTION
**Pull request by chriseth**
Depends on https://github.com/ethereum/solidity/pull/2989

### CODE REVIEW

File: test/boostTest.cpp [hunk 1]
```diff
@@ -63,6 +63,12 @@ test_suite* init_unit_test_suite( int /*argc*/, char* /*argv*/[] )
 			master.remove(id);
 		}
 	}
+	if (dev::test::Options::get().disableSMT)
+	{
+		auto id = master.get("SMTChecker");
+		assert(id != INV_TEST_UNIT_ID);
+		master.remove(id);
```
Comment (axic) [inline test/boostTest.cpp]: Could have a helper for removing a test suite (used by this and `disableIPC`).

**Review by axic — APPROVED** [review]
Apart from the helper comment.

**Comment by chriseth** [comment]
Ah, I think I still have to un-disable actually running the SMT tests.

**Comment by chriseth** [comment]
Disabled SMT tests on windows, I hope this is ready to merge now.

**Pull request merged**